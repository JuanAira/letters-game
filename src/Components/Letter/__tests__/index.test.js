import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { Letter } from '../index';
import SharedProps from '../../../context/SharedProps';

describe('<Letter /> component', () => {
  const handleChangeWord = jest.fn();
  
  const setup = ({ letter = 'A', i= 0 }) => {
    const sharedProps = {
      keys: [1, 2, 3],
      handleChangeWord,
      nextPossibleValues: [],
    };

    const component = render(
      <SharedProps.Provider value={sharedProps} >
        <Letter
          letter={letter}
          i={i}
        />
      </SharedProps.Provider>
    );

    return component;
  };

  it('should match snapshot - no include key', () => {
    const { container } = setup({});
    expect(container).toMatchSnapshot();
  });

  it('should call handleChangeWord when user press a click - include key', () => {
    setup({ i: 1 });
    const letter = screen.getByText(/A/i);
    fireEvent.click(letter);
    
    expect(handleChangeWord).toHaveBeenCalledTimes(1);
  });
});
