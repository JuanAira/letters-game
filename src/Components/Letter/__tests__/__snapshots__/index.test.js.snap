// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Letter /> component should match snapshot - no include key 1`] = `
<div>
  <h3
    class="letter 0 letter--disabled"
  >
    A
  </h3>
</div>
`;
